tests/zfs-tests/tests/functional/xattr/xattr_zfs_zreget.ksh

```diff
@@ -0,0 +1,82 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2019 Datto, Inc. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# See ZoL issues #5866 and #8858.  This test will ensure the fixes for
+# these specific issues have no regression.
+#
+# Exercise the functions zfs_suspend_fs / zfs_resume_fs / zfs_rezget.
+# The test will ensure that we reach zfs code that compares file
+# generation numbers and fetches extended attributes.
+#
+# STRATEGY:
+# 1. Import a pool that has txg number > 2^32.
+# 2. Take snapshot and do send with an online receive.
+# 3. EIO is expected when accessing the received file-system.
+# 4. Access extended attribute.
+# 5. Take second snapshot, write some data.
+# 6. Rollback.
+# 7. Access the resulting file-system.
```

Comment:
Should we be doing some more extensive validation here.  I believe the unwritten assumption here is that we'll see some kind of IO error if this were to regression.  It would be helpful to be explicit about what the expected failure would be.

Reply:
When we hit this at Datto every attempted access was met with EIO.  I added some comments about how it previously would fail.